src/Compilers/CSharp/Portable/Lowering/DiagnosticsPass_ExpressionTrees.cs

```diff
@@ -211,6 +211,19 @@ public override BoundNode VisitCompoundAssignmentOperator(BoundCompoundAssignmen
             }
         }
 
+        public override BoundNode Visit(BoundNode node)
+        {
+            if (_inExpressionLambda &&
+                !(node is BoundConversion) &&
```

Comment:
Why exclude `BoundConversion`?

Reply:
This prevents redundant diagnostics in the [`RefStructDefaultExpressionTree`](https://github.com/dotnet/roslyn/pull/30871/commits/a080a34e149ee13c5b5188e0eb19982982ee8536#diff-609b3f09c132215b9994e189fd2a5861R1572) test case. In that case we'd have a `BoundConversion` containing a `BoundDefaultExpression`, where the type of both nodes `IsByRefLikeType`.

I'm wondering if there are other scenarios that would generate redundant diagnostics. Is there a better practice for handling this kind of problem?

Reply:
We really don't -- cascading diagnostics are handled on a case-by-case basis right now. In theory it sounds like what you really want is only one instance of this diagnostic to be reported per syntax node, but that sounds way too complicated for this bug. I think your solution is fine, but I would add a comment about why you're filtering conversions.